src/odemis/acq/stitching/_registrar.py

```diff
@@ -580,6 +581,11 @@ def _insert_tile_to_grid(self, tile):
                     self.tiles[i].append(None)
                     self.shifts_hor[i].append(None)
                     self.shifts_ver[i].append(None)
+        elif almost_equal(abs(ver_diff), abs(hor_diff)) and abs(hor_diff) > 0 and abs(ver_diff) > 0:
```

Comment:
I don't think almost_equal will work here. `ver_diff` and `hor_diff` might still be a few pixels apart in this case, so you want to test whether the magnitude is approximately the same. I think that's a bit difficult to do and might be more confusing than helpful.

The docstring of `.addTile` explicitly says that diagonal moves are not supported, so we probably don't need to check it here.

Reply:
I increased the tolerance to 1%. I don't actually think it would be such a big problem because in any other case when ver_diff != hor_diff the code would go into the other if statements.